src/dict/sounddir.cc

```diff
@@ -438,7 +439,24 @@ vector< sptr< Dictionary::Class > > makeDictionaries( Config::SoundDirs const &
 
     string indexFile = indicesDir + dictId;
 
-    if ( Dictionary::needToRebuildIndex( dictFiles, indexFile ) || indexIsOldOrBad( indexFile ) ) {
+    // Check if the soundDir and its subdirs' modification date changed, that means the user modified the sound files inside
+
+    bool soundDirModified = false;
+    {
+      QDateTime indexFileModifyTime = QFileInfo( QString::fromStdString( indexFile ) ).lastModified();
+      QDirIterator it( dir.path(),
+                       QDir::AllDirs | QDir::NoDotAndDotDot | QDir::NoSymLinks,
```

Comment:
The NoDotAndDotDot is QDirIterator detail, `/dir/..` is a valid dir in unix :sweat_smile: 

Sound dir doesn't chase SymLinks, so filter them out too.